### PR TITLE
DB-3605: Grid Component

### PR DESCRIPTION
**Pull request by mitchellmarkoff**
## What changes were made?
A Grid component to be used in the nextjs starters was added to the nextjs-kit.

## Where were the changes made?
nextjs-kit

## How have the changes been tested?
The new Grid component was implemented locally and tested with all of our current datasets that get placed into a Grid.
## Additional information

Don't forget to
[add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset)
if needed!

### CODE REVIEW

**Review by CobyPear — CHANGES_REQUESTED** [review]

File: packages/nextjs-kit/tailwind.config.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
 	content: ['./src/components/**/*.tsx'],
+	safelist: [
+		{
+			pattern: /grid-cols-([0-9])/,
+			variants: ['lg'],
+		},
+	],
```
Comment (CobyPear) [inline packages/nextjs-kit/tailwind.config.js:9]: I'm skeptical that this is the best solution to the cols problem, the docs say this is a last resort kind of thing. Is there truly no better way?

File: packages/nextjs-kit/src/components/grid.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * @returns A Higher Order Component that returns the data mapped to the Component in a grid
+ */
+export const withGrid = (Component: React.ElementType) => {
+	const GridedComponent = ({
+		data,
+		FallbackComponent,
+		cols,
+		...props
+	}: {
+		data: Record<string, string | number>[];
```
Comment (CobyPear) [inline packages/nextjs-kit/src/components/grid.tsx:44]: Let's use a generic here

**Review by CobyPear — CHANGES_REQUESTED** [review]

File: packages/nextjs-kit/tailwind.config.js (hunk 1 above)
Comment (mitchellmarkoff) [reply packages/nextjs-kit/tailwind.config.js:9]: I wrestled with this for awhile and have yet to find a solution I think is better. I tried the helper function route outlined [here](https://stackoverflow.com/a/72268889) but that did not fix the problem. The only other way I found to get the dynamic columns working was by side stepping tailwind and adding an inline style like so
```
style={{
	gridTemplateColumns: `repeat(${cols}, minmax(0, 1fr))`,
}}
```
This solution didn't feel any better right and makes it more difficult to add a media query for breakpoint definitions.

Let me know what you think 👍

File: packages/nextjs-kit/src/__tests__/snapshotTests/__snapshots__/grid.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      </h1>
+      <p>
+        Fourth item in the grid
+      </p>
+    </div>
+  </div>
+</DocumentFragment>
+`;
+
+exports[`<Grid /> > should render FallbackComponent instead of 'Grid' 1`] = `
```
Comment (CobyPear) [inline packages/nextjs-kit/src/__tests__/snapshotTests/__snapshots__/grid.test.tsx.snap:44]: Looks like this is rendering the grid?

File: packages/nextjs-kit/tailwind.config.js [hunk 4]
```diff
@@ -1,6 +1,12 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
 	content: ['./src/components/**/*.tsx'],
+	safelist: [
+		{
+			pattern: /grid-cols-([0-9])/,
```
Comment (CobyPear) [inline packages/nextjs-kit/tailwind.config.js:6]: I think we need the prefix here
```suggestion
			pattern: /ps-grid-cols-([0-9])/,
```

**Review by CobyPear — CHANGES_REQUESTED** [review]

File: packages/nextjs-kit/tailwind.config.js (hunk 4 above)
Comment (mitchellmarkoff) [reply packages/nextjs-kit/tailwind.config.js:6]: Sorry I forgot to add a comment, I tested without the prefix and it seems that it is not necessary.

File: .changeset/wet-foxes-matter.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@pantheon-systems/nextjs-kit': patch
```
Comment (CobyPear) [inline .changeset/wet-foxes-matter.md:2]: I think this can be a minor too given your reply on #303

**Pull request merged**